segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessorTest.java

```diff
@@ -1506,6 +1521,128 @@ public void testReadTableEntriesDelta() throws Exception {
         assertFalse(results.contains(e5));
     }
 
+    @Test
+    public void testCreateSealTruncateDeleteIndexSegment() throws Exception {
+        String segment = "testCreateSealTruncateDeleteIndexSegment";
+        String indexSegment = getIndexSegmentName(segment);
+        ByteBuf data = Unpooled.wrappedBuffer("Hello world\n".getBytes());
+        ServiceBuilder serviceBuilder = newInlineExecutionInMemoryBuilder(getBuilderConfig());
+
+        serviceBuilder.initialize();
+        StreamSegmentStore store = serviceBuilder.createStreamSegmentService();
+        @Cleanup
+        EmbeddedChannel channel = createChannel(store);
+        WireCommands.SegmentCreated created = (WireCommands.SegmentCreated) sendRequest(channel, new WireCommands.CreateSegment(1, segment, WireCommands.CreateSegment.NO_SCALE, 0, "", 1024L));
+        assertEquals(segment, created.getSegment());
+
+        UUID uuid = UUID.randomUUID();
+        WireCommands.AppendSetup setup = (WireCommands.AppendSetup) sendRequest(channel, new WireCommands.SetupAppend(2, uuid, segment, ""));
+
+        assertEquals(segment, setup.getSegment());
+        assertEquals(uuid, setup.getWriterId());
+
+        data.retain();
+
+        //Append 40 bytes of data
+        sendRequest(channel, new Append(segment, uuid, 1, new WireCommands.Event(data), 1L));
+        sendRequest(channel, new Append(segment, uuid, 2, new WireCommands.Event(data), 1L));
+
+        //Total length
+        assertEquals(store.getStreamSegmentInfo(segment, PravegaRequestProcessor.TIMEOUT).join().getLength(), 40);
+
+        //Validating the data append data
+        WireCommands.SegmentRead result = (WireCommands.SegmentRead) sendRequest(channel, new WireCommands.ReadSegment(segment, 0, 20, "", 1L));
+        ByteBuf resultAfterOneAppend = result.getData();
+        assertEquals("Hello world", resultAfterOneAppend.toString(Charset.defaultCharset()).trim());
+
+        // Truncate one event
+        final long truncateOffset = 20;
+
+        //Before truncation validation
+        assertEquals(0, store.getStreamSegmentInfo(segment, PravegaRequestProcessor.TIMEOUT).join().getStartOffset());
+        assertEquals(0, store.getStreamSegmentInfo(indexSegment, PravegaRequestProcessor.TIMEOUT).join().getStartOffset());
+
+        AssertExtensions.assertGreaterThan("Nothing to truncate.", 0, truncateOffset);
+        //After truncation validation
+        sendRequest(channel, new WireCommands.TruncateSegment(requestId, segment, truncateOffset, ""));
+
+        assertEquals(truncateOffset, store.getStreamSegmentInfo(segment, PravegaRequestProcessor.TIMEOUT).join().getStartOffset());
+        assertEquals(0, store.getStreamSegmentInfo(indexSegment, PravegaRequestProcessor.TIMEOUT).join().getStartOffset());
```

Comment:
I think its not a proper test, indexSegment index should also be greater than 0.

Reply:
Here, offset of the index segment after calling I**ndexRequestProcessor.locateOffsetForIndexSegment** its returns as zero so truncation has not happened so the starting offset for the index segment is zero.

Reply:
Got it, Thanks